sv.c

```diff
@@ -9608,7 +9608,26 @@ Perl_newSVnv(pTHX_ const NV n)
     SV *sv;
 
     new_SV(sv);
-    sv_setnv(sv,n);
+    /* Inlining ONLY the small relevant subset of sv_setnv here
+     * for performance. */
+
+    /* We're starting from SVt_FIRST, so provided that's
+     * actual 0, we don't have to unset any SV type flags
+     * to promote to SVt_IV. */
+    STATIC_ASSERT_STMT(SVt_FIRST == 0);
```

Comment:
All this is fine, but I wonder if it is necessary: if we set the type below instead with:
```
  SvFLAGS(sv) = SVt_NV;  /* and optionally ... | SVf_NOK | SVp_NOK */
```
.. then I don't think anything remains reliant on SVt_FIRST does it?

+1 from me to merge this in any case.

Reply:
That comment - which contains a copy&paste error that I'll fix in a sec - and `STATIC_ASSERT_STMT` are only there because they're in `Perl_newSViv`. I figured it would be nice to have visual consistency, but don't think it's necessary to have it. I'm not fussed whether it stays or goes.

Reply:
Now that `newSV_type` is inlined, it's cleaner to just use that, so I've updated the PR to do so.

Reply:
I think it would make sense to change `Perl_newSViv` to use `newSV_type` too. It would make it a bit simpler and I expect no change in the performance.

Reply:
Agreed. I'll roll that into this PR shortly.